server/src/utils/asset.util.ts

```diff
@@ -14,13 +14,14 @@ import { IBulkAsset, ImmichFile, UploadFile, UploadRequest } from 'src/types';
 import { checkAccess } from 'src/utils/access';
 
 export const getAssetFile = (files: AssetFile[], type: AssetFileType | GeneratedImageType) => {
-  return files.find((file) => file.type === type);
+  return files ? files.find((file) => file.type === type) : null;
```

Comment:
Either the type is wrong, or `files` is always defined and you don't need the ternary

Reply:
The method signature and implementation don't make sense together though. Either you assume the method signature, in which case you remove the ternary, or you assume files can be undefined/null/whatever, in which case the signature should represent that

Reply:
I got this fixed now, only needed a little conditional in another place instead of here.